### PR TITLE
libcephfs: define AT_NO_ATTR_SYNC back for backward compatibility

### PR DESCRIPTION
**Pull request by lxbsz**
This was introduce by commit e2a67f2a65553ad45721bb391081bc61aa97e0e9,
for the third part applications they may still use the old macro.

Add it back and marked it as deprecated.

Fixes: https://tracker.ceph.com/issues/56638
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by mchangir — APPROVED** [review]
looks good to me

File: PendingReleaseNotes [hunk 1]
```diff
@@ -46,6 +46,8 @@
 * RGW: The beast frontend's HTTP access log line uses a new debug_rgw_access
   configurable. This has the same defaults as debug_rgw, but can now be controlled
   independently.
+* The 'AT_NO_ATTR_SYNC' macro is deprecated and please to use standard 'AT_STATX_DONT_SYNC'
```
Comment (phlogistonjohn) [inline PendingReleaseNotes:49]: Grammar:  The 'AT_NO_ATTR_SYNC' macro is deprecated, please use the standard 'AT_STATX_DONT_SYNC' macro. The 'AT_NO_ATTR_SYNC' macro will be removed in the future.

Comment (lxbsz) [reply PendingReleaseNotes]: Looks better and done. Thanks!

**Review by kotreshhr — APPROVED** [review]

**Comment by dparmar18** [comment]
LGTM

**Review by dparmar18 — APPROVED** [review]

**Review by ajarr — APPROVED** [review]

**Review by nmshelke — APPROVED** [review]
LGTM

**Review by vshankar — APPROVED** [review]

**Comment by phlogistonjohn** [comment]
Hi all, I read on the mailing list that release planning for the next quincy has started. Could I please ask that this PR be expedited along with related backports? The earlier change that removed the macro was already backported (to quincy and pacific IIRC) so if quincy is released without a backport for this change as well it could potentially break existing code.  Thanks!

**Comment by vshankar** [comment]
> Hi all, I read on the mailing list that release planning for the next quincy has started. Could I please ask that this PR be expedited along with related backports? The earlier change that removed the macro was already backported (to quincy and pacific IIRC) so if quincy is released without a backport for this change as well it could potentially break existing code. Thanks!

This is in my test branch for running teuthology tests and should get merged soon.

**Comment by lxbsz** [comment]
Resolved the conflict in `PendingReleaseNotes`.

**Comment by vshankar** [comment]
@lxbsz could you rebase this please. I hope that fixes the make check and other errors...

**Comment by vshankar** [comment]
jenkins retest this please

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Aug-22

**Pull request merged**